### PR TITLE
Refactor blobstore filesystem tests for Mac OS X

### PR DESCRIPTION
**Pull request by jdaggett**
This PR refactors the filesystem blobstore tests slightly for Mac OS.

- Utilizes class/method level `dataProvider` and `dataProviderClass` in tests where necessary
- (minor) 3-space indent cleanup

`org.jclouds.util.TestUtils`
- pulled up constants from `org.jclouds.filesystem.utils.TestUtils`
- pulled up `@DataProvider` for Mac OS/Windows
- pulled up `isWindowsOs()` method

`org.jclouds.filesystem.integration.FilesystemBlobIntegrationTest`
- pulled up description of Mac OS xattrs bug info to class level
- added failing methods from `BaseBlobIntegrationTest`
- now skips all tests on Mac OS

@zack-shoylev Can you please run this on Windows for a sanity check? Thx!

### CODE REVIEW

**Comment by andrewgaul** [comment]
@kahing Can you review this?

File: apis/filesystem/src/test/java/org/jclouds/filesystem/integration/FilesystemBlobIntegrationTest.java [hunk 1]
```diff
@@ -41,38 +52,65 @@ protected Properties setupProperties() {
...
+   @Test(dataProvider = "ignoreOnMacOSX")
+   @Override
+   public void testPutByteSource() throws Exception {
+      super.testPutByteSource();
+   }
+
+   @Test(dataProvider = "ignoreOnMacOSX")
+   @Override
+   public void testPutInputStream() throws Exception {
+      super.testPutInputStream();
```
Comment (andrewgaul) [inline apis/filesystem/src/test/java/org/jclouds/filesystem/integration/FilesystemBlobIntegrationTest.java]: This doesn't seem like the right approach.  Instead this should override `checkContentMetadata` and similar.

**Comment by andrewgaul** [comment]
There's a lot going on here -- can we break this up into a few different pull requests, e.g., reindentation, Mac OS fixes, etc.?  Thanks!

**Comment by jdaggett** [comment]
@andrewgaul sure thing. closing...

**Comment by jdaggett** [comment]
@kahing @andrewgaul Just FYI I didn't spend any more time on this after closing this PR because of other priorities. As of today, the master branch has been broken for a week on Mac OS X. :disappointed: Can you guys please provide a fix for it that makes sense for the filesystem provider? Should I submit a JIRA?

**Comment by kahing** [comment]
sure thing I will look at this now

**Pull request closed without merging**